--- ./wg-quick	2026-06-20 14:10:13.668784700 +1200
+++ ./wg-quick-v1.0.20260223	2026-06-20 14:13:13.956232300 +1200
@@ -238,11 +238,13 @@
 	printf -v restore '%sCOMMIT\n*mangle\n-I POSTROUTING -m mark --mark %d -p udp -j CONNMARK --save-mark %s\n-I PREROUTING -p udp -j CONNMARK --restore-mark %s\nCOMMIT\n' "$restore" $table "$marker" "$marker"
 	printf -v nftcmd '%sadd rule %s %s postmangle meta l4proto udp mark %d ct mark set mark \n' "$nftcmd" "$pf" "$nftable" $table
 	printf -v nftcmd '%sadd rule %s %s premangle meta l4proto udp meta mark set ct mark \n' "$nftcmd" "$pf" "$nftable"
-	[[ $proto == -4 ]] && [[ $(sysctl -n net.ipv4.conf.all.src_valid_mark) -ne 1 ]] && cmd sysctl -q net.ipv4.conf.all.src_valid_mark=1
+	[[ $proto == -4 ]] && set +e && [[ $(sysctl -n net.ipv4.conf.all.src_valid_mark) -ne 1 ]] && sysctl -q net.ipv4.conf.all.src_valid_mark=1; set -e
 	if type -p nft >/dev/null; then
 		cmd nft -f <(echo -n "$nftcmd")
 	else
+		[ "$ALLOW_MISSING_IPTABLES_RULES" = "1" ] && set +e
 		echo -n "$restore" | cmd $iptables-restore -n
+		[ "$ALLOW_MISSING_IPTABLES_RULES" = "1" ] && set -e
 	fi
 	HAVE_SET_FIREWALL=1
 	return 0
